### PR TITLE
Work in progress - reveng using 2.2

### PR DESCRIPTION
**Pull request by ErikEJ**
@Herdo FYI

### CODE REVIEW

File: src/GUI/EFCorePowerTools.Shared/Models/TableInformationModel.cs [hunk 1]
```diff
@@ -71,14 +71,34 @@ public bool HasPrimaryKey
         /// Gets the unsafe (unescaped) full name of the table.
         /// </summary>
         [IgnoreDataMember]
-        public string UnsafeFullName => $"{Schema}.{Name}";
+        public string UnsafeFullName
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools.Shared/Models/TableInformationModel.cs]: @Herdo  !

File: src/GUI/EFCorePowerTools.Shared/Models/TableInformationModel.cs [hunk 2]
```diff
@@ -71,14 +71,34 @@ public bool HasPrimaryKey
...
+            }
+        }
 
         /// <summary>
         /// Gets the safe (escaped) full name of the table.
         /// </summary>
         [IgnoreDataMember]
-        public string SafeFullName => $"[{Schema}].[{Name}]";
-
+        public string SafeFullName
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools.Shared/Models/TableInformationModel.cs]: @Herdo

File: src/GUI/EFCorePowerTools.Shared/Models/TableInformationModel.cs [hunk 3]
```diff
@@ -112,12 +130,19 @@ public static TableInformationModel Parse(string table)
...
 
-            var split = table.Split('.');
-            if (split.Length != 2)
-                throw new ArgumentException(@"Value cannot be parsed.", nameof(table));
+            string schema = null;
+            string name = null;
 
-            var schema = split[0];
-            var name = split[1];
+            var split = table.Split('.');
```
Comment (ErikEJ) [inline src/GUI/EFCorePowerTools.Shared/Models/TableInformationModel.cs]: @Herdo

File: src/GUI/UnitTests/Models/TableInformationModelTest.cs [hunk 4]
```diff
@@ -67,13 +67,14 @@ public void Parse_ArgumentException_NotTwoPeriods()
         {
             // Arrange
             var table1 = "Album";
+            //TODO dbo.Test.Table is a valid table name, must update parser!
```
Comment (ErikEJ) [inline src/GUI/UnitTests/Models/TableInformationModelTest.cs]: @Herdo

**Comment by Herdo** [comment]
Now this is making things complicated.
So we either drop the information completely, and just hold the name again, like before the factoring, or we come up with an intelligent way to solve this.

`dbo.Test` would be a table for MS SQL and SQL CE.
`dbo.Test.Table` would be an error for MS SQL, but a table called `[dbo.Test.Table]` for SQL CE.

Would this work:
`A` --> parse A to table
`A.B` --> parse A to schema and B to table
`A.B.C` --> parse A.B.C to table
`A.B.C.D` --> parse A.B.C.D to table
.......

**Comment by ErikEJ** [comment]
I vote for dropping schema for now - will update this PR

**Comment by Herdo** [comment]
@ErikEJ Removing the schema _might_ require changes to the `ReverseEngineerOptions.cs` reading methods.

**Pull request merged**